paasta_tools/cli/cmds/check.py

```diff
@@ -198,95 +198,81 @@ def yaml_check(service_path):
+        found_yaml = True
     if not found_yaml:
         paasta_print(PaastaCheckMessages.YAML_MISSING)
 
 
-def get_chronos_steps(service, soa_dir):
-    """This is a kind of funny function that gets all the chronos instances
-    for a service and massages it into a form that matches up with what
+def get_deploy_groups_used_by_framework(instance_type, service, soa_dir):
+    """This is a kind of funny function that gets all the instances for specified
+    service and framework, and massages it into a form that matches up with what
     deploy.yaml's steps look like. This is only so we can compare it 1-1
-    with what deploy.yaml has for linting."""
-    steps = []
-    for cluster in list_clusters(service, soa_dir):
-        for _, instance in get_service_instance_list(
-                service=service,
-                cluster=cluster,
-                instance_type='chronos',
-                soa_dir=soa_dir,
-        ):
-            config = load_chronos_job_config(
-                service=service,
-                instance=instance,
-                cluster=cluster,
-                soa_dir=soa_dir,
-                load_deployments=False,
-            )
-            steps.append(config.get_deploy_group())
-    return steps
+    with what deploy.yaml has for linting.
 
+    :param instance_type: one of 'marathon', 'chronos', 'adhoc'
+    :param service: the service name
+    :param soa_dir: The SOA configuration directory to read from
+
+    :returns: a list of deploy group names used by the service.
+    """
 
-def get_marathon_steps(service, soa_dir):
-    """This is a kind of funny function that gets all the marathon instances
-    for a service and massages it into a form that matches up with what
-    deploy.yaml's steps look like. This is only so we can compare it 1-1
-    with what deploy.yaml has for linting."""
     steps = []
     for cluster in list_clusters(service, soa_dir):
         for _, instance in get_service_instance_list(
                 service=service,
                 cluster=cluster,
-                instance_type='marathon',
+                instance_type=instance_type,
                 soa_dir=soa_dir,
         ):
-            config = load_marathon_service_config(
-                service=service,
-                instance=instance,
-                cluster=cluster,
-                soa_dir=soa_dir,
-                load_deployments=False,
-            )
-            steps.append(config.get_deploy_group())
+            try:
+                config = get_instance_config(
+                    service=service,
+                    instance=instance,
+                    cluster=cluster,
+                    soa_dir=soa_dir,
+                    load_deployments=False,
+                    instance_type=instance_type,
+                )
+                steps.append(config.get_deploy_group())
+            except NotImplementedError:
+                pass
     return steps
```

Comment:
this is our chance to "get this right", please call these "deploy_groups", as that is what they are.